### ISSUE TITLE
Message is empty

### ISSUE THREAD

**Issue by lunascat** [jenkinsci/slack-plugin#476]
We are using the slack-plugin with RocketChat.
After configuring the plugin in Jenkins and testing the connection, our RocketChat bot successfully posts a message, but this message is empty! Same occurs, when we finally use the plugin in one of our pipelines. The problem occurs no matter, if we use a custom message or not.

When posting a new message directly to RocketChat via curl, everything works as expected:
`curl -d '{"text": "Hello"}' -H 'Content-Type: application/json' -X POST 'https://rocketchat.xxx.de/hooks/token'`

We tried version 2.9 and 2.12.

**Comment by lunascat** [comment]
Any news on this?

**Comment by timja** [comment]
Sorry I don't use rocket chat, no idea, contributions welcome to fix it and add tests to cover it

**Comment by casz** [comment]
Yup, you could even have a container test: https://hub.docker.com/_/rocket-chat

**Comment by cinlloc** [comment]
Same kind of problem with Mattermost at sending data: 
```Sep 04, 2019 1:56:04 PM WARNING jenkins.plugins.slack.StandardSlackService publish
Slack post may have failed. Response: null
Sep 04, 2019 1:56:04 PM WARNING jenkins.plugins.slack.StandardSlackService publish
Response Code: 405
```

**Comment by cinlloc** [comment]
Workaround for RocketChat: token has to be removed from `Override url` advanced parameter in job configuration. E.g.:
```
Override url: https://rocketchat.xxx.de/hooks/
Credential: <select a secret with token value>
```
If not, Slack plugin add the token at the end of the POST request, which results in 404 error.

**Comment by casz** [comment]
@cinlloc pretty sure that is not a workaround. That is the intended use of the plugin.
